src/ui/modals/smithing-modal-window.js

```diff
@@ -0,0 +1,156 @@
+            .setOrigin(0, 0)
+            .setDepth(2);
+        
+        this.window.visible = this.visible;
+
+        this.exitButton = new Button(this.scene, this.x + 473, this.y + 8, 19, 19);
+        this.exitButton.on("pointerup", () => {
+            if (this.visible) {
+                this.setVisible(false);
+            }
+        });
+    }
+
+    clearChoices() {
+        for (let element of this.elements) {
+            element.destroy();
+        }
+        
+        this.elements = [];
+    }
+
+    getSpriteName(name) {
+        let result = name.toLowerCase();
+        result.toLowerCase();
+        result = result.replaceAll(" ", "-");
+        return result;
+    }
+
+    async setChoices(bar, barSupply, smithingLevel) {
+        this.clearChoices();
+
+        let elements = [];
+
+        switch (bar) {
+            case "Bronze Bar":
+                    elements = ["BronzeDagger", "BronzeSword", "SKIP", "BronzeScimitar", "BronzeAxe", "Bronze2hSword"];
+                break;
+            default:
+                return;
+        }
+
+        let rowIndex = 0;
+
+        let positionX = this.x + this.leftOffset + this.horizontalPadding;
+        let positionY = this.y + this.topOffset + this.verticalPadding;
+
+        for (let element of elements) {
+            if (element === "SKIP") {
+                positionX += this.iconWidth + this.horizontalPadding;
+                rowIndex += 1;
+                continue;
+            }
+
+            if (rowIndex >= this.maxIconsPerRow) {
+                positionX = this.x + this.leftOffset + this.horizontalPadding;
+                positionY += this.iconHeight + this.verticalPadding + 24;
+                rowIndex = 0;
+            }
+
+            const item = await getItemClass(element, this.scene.window);
+            const spriteName = await this.getSpriteName(item.name);
+
+            const barFlag = barSupply >= item.bars[0].count ? true : false;
+            const levelFlag = smithingLevel >= item.smithingLevel ? true : false;
+
+            if (barFlag && levelFlag) {
+                let elementButton = new Button(this.scene, positionX, positionY, this.buttonWidth, this.buttonHeight);
+                elementButton.on("pointerup", () => {
+                    if (this.visible) {
+                        this.choice = element;
+                        this.setVisible(false);
+                    }
+                });
+
+                this.elements.push(elementButton);
+            }
+
+            let elementIcon = this.scene.add
+                .image(positionX + 12, positionY, spriteName)
+                .setOrigin(0, 0)
+                .setDepth(3);
+            elementIcon.visible = this.visible
+
+            const textFont = levelFlag ? FONTS.SMITH_UNLOCKED : FONTS.SMITH_LOCKED;
+
+            let elementText = this.scene.add.text(positionX, positionY + this.iconHeight, item.item, textFont).setDepth(3);
```

Comment:
I wonder if you can chain a `setVisible(this.visible)` on the end so it can be all in one